### DISCUSSION TITLE
diagnosticMode in lsp

### DISCUSSION THREAD

**Discussion by korchizhinskiy** [microsoft/pyright#9609] [General]
Hi. Help me, please, with configuration of LSP pyright.
When I working, pyright lsp helping with auto-import, autocomplete and etc with known files (which I opened recently in current nvim session). I read in issues, that param diagnosticMode in "workspace" value can scan all project on start and can help me without manual opening all files, which need me for suggestions of finding imports.
But, when I set this param (on screenshot), nothing changed. What's wrong?
In my project I have only pyproject.toml file (not pyrightconfig.json etc.)
![Image](https://github.com/user-attachments/assets/a0485ba7-5b27-4955-9b9d-4aa3e0cb2c21)

Pyright version - 1.1.389

**Comment by erictraut** [comment]
Converting to a discussion topic since this isn't a bug report or a feature request.

**Comment by erictraut** [comment]
Yes, when you set `diagnosticMode` to "workspace", pyright will report diagnostics for all files within your project. By default, `diagnosticMode` is set to "openFilesOnly", and pyright will report diagnostics only for files that are currently open in your editor. 

I use the "workspace" mode frequently within VS Code, so I know that it works. I don't have any experience with nvim, and I don't know how or where it reports diagnostics for files that are not currently opened. In VS Code, you'll see these in the "Problems" tab. If you are not seeing diagnostics reported, you may need to find someone who knows about nvim and how to configure it.

**Reply by korchizhinskiy** [reply]
Ok, thanks, it is my bad, that I want to config this param in pyproject. I think, i have to change settings of lsp config in nvim.

**Reply by erictraut** [reply]
Ah yes, you're correct. The `python.analysis.diagnosticMode` is an LSP setting, so it doesn't go in pyproject.

**Reply by korchizhinskiy** [reply]
I am using Astrovim configuration for Nvim, and now send, how I decided this problem.
https://github.com/neovim/nvim-lspconfig/blob/master/doc/configs.md#pyright
This link for common explanation of pyright configuration as lsp

But in Astrovim i set this settings.
```lua
return {
  "AstroNvim/astrolsp",
  ---@type AstroLSPOpts
    config = {
      pyright = {
        settings = {
          python = {
            analysis = {
              diagnosticMode = "workspace",
            },
          },
        },
        -- clangd = { capabilities = { offsetEncoding = "utf-8" } },
      },
    },
...
```

This resolve my problem and project scan on start nvim session. 
Thank you!

May be this message will help anyone :)
